scripts/Engineering/gui/engineering_diffraction/tabs/focus/model.py

```diff
@@ -36,77 +42,114 @@ def focus_run(self, sample_paths, banks, plot_output, instrument, rb_num, spectr
+                    DeleteWorkspace(sample_ws_clone)
+                    DeleteWorkspace(curves_ws_clone)
+                output_workspaces.append(workspaces_for_run)
                 self._output_sample_logs(instrument, run_no, sample_workspace, rb_num)
+
         # Plot the output
         if plot_output:
             for ws_names in output_workspaces:
                 self._plot_focused_workspaces(ws_names)
 
     @staticmethod
     def _run_focus(input_workspace,
-                   output_workspace,
+                   tof_output_name,
                    vanadium_integration_ws,
                    vanadium_curves_ws,
-                   bank,
-                   full_calib_ws=None,
-                   spectrum_numbers=None):
-        kwargs = {
-            "InputWorkspace": input_workspace,
-            "OutputWorkspace": output_workspace,
-            "VanIntegrationWorkspace": vanadium_integration_ws,
-            "VanCurvesWorkspace": vanadium_curves_ws
-        }
-        if full_calib_ws is not None:
-            kwargs["DetectorPositions"] = full_calib_ws
-
-        if bank is None:
-            kwargs["SpectrumNumbers"] = spectrum_numbers
+                   df_kwarg,
+                   full_calib,
+                   region_calib):
+        if input_workspace.getRun().getProtonCharge() > 0:
+            NormaliseByCurrent(InputWorkspace=input_workspace, OutputWorkspace=input_workspace)
         else:
-            kwargs["Bank"] = bank
-
-        try:
-            return EnggFocus(**kwargs)
-        except RuntimeError as e:
-            logger.error(
-                "Error in focusing, Could not run the EnggFocus algorithm successfully for bank "
-                + str(bank) + ". Error Description: " + str(e))
-            raise RuntimeError()
+            logger.warning(f"Skipping focus of run {input_workspace.name()} because it has invalid proton charge.")
+            return False
+        input_workspace /= vanadium_integration_ws
+        ReplaceSpecialValues(InputWorkspace=input_workspace, OutputWorkspace=input_workspace, NaNValue=0,
+                             InfinityValue=0)
+        ApplyDiffCal(InstrumentWorkspace=input_workspace, CalibrationWorkspace=full_calib)
+        ws_d = ConvertUnits(InputWorkspace=input_workspace, Target='dSpacing')
+        focused_sample = DiffractionFocussing(InputWorkspace=ws_d, **df_kwarg)
+        curves_rebinned = RebinToWorkspace(WorkspaceToRebin=vanadium_curves_ws, WorkspaceToMatch=focused_sample)
+        normalised = Divide(LHSWorkspace=focused_sample, RHSWorkspace=curves_rebinned,
+                            AllowDifferentNumberSpectra=True)
+        ApplyDiffCal(InstrumentWorkspace=normalised, CalibrationWorkspace=region_calib)
+        dspacing_output_name = tof_output_name + "_dSpacing"
+        CloneWorkspace(InputWorkspace=normalised, OutputWorkspace=dspacing_output_name)
```

Comment:
Is this not actually intended to be a Rename rather than Clone - we may nbot need either if we call it by the correct name above by specifying `OutputWorkspace` in `DiffractionFocusing `

Reply:
There are maybe a few more places we could cut down on the number of workspaces by specifying the OutputWorkspace - I will have another look.

Reply:
Did you mean ApplyDiffCal rather than DiffractionFocusing? if so that's an in place op so doesn't take an output workspace, but I think we can do what you had in mind by just renaming `normalised`

Reply:
I had a separate comment above related to the divide being in-place such that `normalised` is still `focused_sample` - so here I am saying can we call `focused_sample` (output of a call to DiffractionFocusing) whatever dspacing_output_name is (i.e. we have OutputWorkspace=tof_output_name+"_dSpacing" on L140 of this file?